src/shogun/labels/MulticlassLabels.cpp

```diff
@@ -184,4 +185,53 @@ CMulticlassLabels* CMulticlassLabels::obtain_from_generic(CLabels* labels)
 	CMulticlassLabels* casted = dynamic_cast<CMulticlassLabels*>(labels);
 	SG_REF(casted)
 	return casted;
-}
\ No newline at end of file
+}
+
+namespace shogun
+{
+	Some<CMulticlassLabels> multiclass_from_multiclass(CMulticlassLabels* orig)
+	{
+		return Some<CMulticlassLabels>::from_raw(orig);
+	}
+
+	Some<CMulticlassLabels> multiclass_from_dense(CDenseLabels* orig)
+	{
+		auto int_labels = orig->get_int_labels();
+		std::set<int32_t> unique(int_labels.begin(), int_labels.end());
+		REQUIRE(
+				(*std::min_element(unique.begin(), unique.end())) == 0 &&
+				(*std::max_element(unique.begin(), unique.end())) == (index_t)unique.size()-1,
+				"Multiclass labels must be contiguous integers in [0, ..., num_classes -1].\n"
```

Comment:
can we fixme/todo this... as this should not be a hard requirement, right?

Reply:
yep!